src/osd/scrubber/osd_scrub_sched.h

```diff
@@ -122,193 +122,483 @@ ScrubQueue interfaces (main functions):
 #include "utime.h"
 
 class PG;
+class PgScrubber;
+class OSDService;
+class ScrubQueue;
+template<>
+struct fmt::formatter<Scrub::SchedTarget>;
 
 namespace Scrub {
 
 using namespace ::std::literals;
 
-// possible outcome when trying to select a PG and scrub it
-enum class schedule_result_t {
-  scrub_initiated,     // successfully started a scrub
-  none_ready,	       // no pg to scrub
-  no_local_resources,  // failure to secure local OSD scrub resource
-  already_started,     // failed, as already started scrubbing this pg
-  no_such_pg,	       // can't find this pg
-  bad_pg_state,	       // pg state (clean, active, etc.)
-  preconditions	       // time, configuration, etc.
+struct scrub_schedule_t {
+  utime_t scheduled_at{};
+  utime_t deadline{0, 0};
```

Comment:
I don't think the intention is to guarantee that the deadlines are met.  Rename this to target.